var f = $() {
	var v;
	try {
		v = 1/0;
	}
	catch(e) {
		throw "catched: " + e;
	}
	finally {
		print("does not executed...\n");
	}
};

try {
	f();
}
catch(e) {
	print("exception: \'" + e + "\'\n");
}
finally {
	print("finally...\n");
}
print("no problems!\n");
